src/main/java/com/bandwidth/sdk/BandwidthClient.java

```diff
@@ -196,6 +200,27 @@ public void setCredentials(final String userId, final String apiToken, final Str
         usersUri = String.format(BandwidthConstants.USERS_URI_PATH, userId);
         this.token = apiToken;
         this.secret = apiSecret;
+
+        validateCredentials();
+    }
+
+    /**
+     * Validate if the credentials are set and has access to catapult
+     */
+    private static void validateCredentials() {
+        try {
+            Account.get().getAccountInfo();
+        } catch (Exception e) {
+            if (e instanceof AppPlatformException) {
+                AppPlatformException appEx = (AppPlatformException) e;
+
+                if (appEx.getStatus() == 401){
```

Comment:
Only error 401 matters here. It refers to this case of error: {"category":"authorization","code":"access-denied","message":"The authenticated user does not have sufficient permissions to access this resource","details":[{"name":"requestMethod","value":"GET"},{"name":"remoteAddress","value":"127.0.0.1"},{"name":"requestPath","value":"/users/your_User_Id_here/account"}]}